### ISSUE TITLE
R2: Apply stopword mask to all loss components

### ISSUE THREAD

**Issue by mateon01** [sa-project-private-repo/opensearch-neural-pre-train#22]
## Problem

In `SPLADELossV26.forward()`, the stopword mask is only applied to the anchor representation for InfoNCE:
```python
masked_anchor = anchor_repr * self.stopword_mask  # Only used for InfoNCE
loss_infonce = self.infonce_loss(masked_anchor, positive_repr, negative_repr)

# These use UNMASKED anchor_repr:
loss_self = self.self_loss(anchor_repr, ...)
loss_positive = self.positive_loss(anchor_repr, ...)
loss_flops = self.flops_loss(anchor_repr)
```

## Impact

- `lambda_self=0.5` + `lambda_positive=2.0` = +2.5 force promoting particle activation
- `lambda_flops_q=3e-4` = -3e-4 force suppressing (via V29 unweighted FLOPS)
- Net imbalance: **8,000x** in favor of particles

## Solution

Apply stopword mask to all loss components:
```python
masked_anchor = anchor_repr * self.stopword_mask
loss_infonce = self.infonce_loss(masked_anchor, positive_repr, negative_repr)
loss_self = self.self_loss(masked_anchor, anchor_input_ids, anchor_attention_mask)
loss_positive = self.positive_loss(masked_anchor, positive_input_ids, positive_attention_mask)
```

Also apply mask to positive/negative representations for document-side losses.

## Priority
Critical - Eliminates the 8,000x force imbalance.

## Related
- #19, #20
